knimin/lib/data_access.py

```diff
@@ -3482,6 +3482,177 @@ def delete_dna_plate(self, dna_plate_id):
                 # plate did not exist
                 raise ValueError("DNA plate %s does not exist" % dna_plate_id)
 
+    def get_dna_plate_list(self):
+        """Gets the list of all dna plates
+
+        Returns
+        -------
+        list of dict
+            {id : int, name : str, date : datetime}
+            Plate id, plate name and date
+        """
+        with TRN:
+            sql = """SELECT dna_plate_id as id, name, created_on::date as date
+                     FROM pm.dna_plate
+                     ORDER BY date DESC"""
+            TRN.add(sql)
+            return [dict(row) for row in TRN.execute_fetchindex()]
+
+    def get_targeted_primer_plates(self):
+        """Returns the list of all targeted primer plates plates
+
+        Returns
+        -------
+        list of dict
+            {id: int, name: str, notes: str, linker_primer_sequence: str
+             target_gene_region: str}
+        """
+        with TRN:
+            sql = """SELECT targeted_primer_plate_id AS id, name, notes,
+                            linker_primer_sequence, target_gene_region
+                     FROM pm.targeted_primer_plate
+                     ORDER BY id"""
+            TRN.add(sql)
+            return [dict(row) for row in TRN.execute_fetchindex()]
+
+    def prepare_targeted_libraries(self, plate_links, email, robot, tm300tool,
+                                   tm50tool, mastermix_lot, water_lot):
+        """Stores the target gene library prep information
+
+        Parameters
+        ----------
+        plate_links : list of dicts
```

Comment:
Note that a DNA plate can be converted into multiple libraries, with same or different prep protocols